parser/index.ts

```diff
@@ -227,17 +238,24 @@ function groupRanges(ranges: Range[]) {
 }
 
 function parseExpensiMarkToRanges(markdown: string): Range[] {
-  const html = parseMarkdownToHTML(markdown);
-  const tokens = parseHTMLToTokens(html);
-  const tree = parseTokensToTree(tokens);
-  const [text, ranges] = parseTreeToTextAndRanges(tree);
-  if (text !== markdown) {
-    // text mismatch, don't return any ranges
+  try {
+    const html = parseMarkdownToHTML(markdown);
+    const tokens = parseHTMLToTokens(html);
+    const tree = parseTokensToTree(tokens);
+    const [text, ranges] = parseTreeToTextAndRanges(tree);
+    if (text !== markdown) {
+      throw new Error(
+        `[react-native-live-markdown] Parsing error: the processed text does not match the original markdown input. This may be caused by incorrect parsing functions or invalid input markdown.`,
+      );
```

Comment:
Can we add the values of text and markdown to the error message
```suggestion
      throw new Error(
        `[react-native-live-markdown] Parsing error: the processed text does not match the original markdown input. This may be caused by incorrect parsing functions or invalid input markdown.`,
        {processed: text, original: markdown}
      );
```

Reply:
Unfortunately, this is not how this works, we would need to create a CustomError extending Error to make it work. 

I picked an easier solution by directly injecting these strings into the error message